crates/sail-python-udf/src/cereal/pyspark_udf.rs

```diff
@@ -1,79 +1,75 @@
-use std::hash::{Hash, Hasher};
+use std::fmt::Debug;
 
-use datafusion_common::{plan_datafusion_err, Result};
+use pyo3::exceptions::PyValueError;
 use pyo3::intern;
 use pyo3::prelude::*;
+use pyo3::sync::GILOnceCell;
 use pyo3::types::PyModule;
 use sail_common::spec;
 
-use crate::cereal::{check_python_udf_version, PythonFunction};
+use crate::cereal::check_python_udf_version;
 use crate::config::SparkUdfConfig;
 use crate::error::{PyUdfError, PyUdfResult};
 
-#[derive(Debug)]
-pub struct PySparkUdfObject(pub PyObject);
+pub struct PySparkUdfObject {
+    data: Vec<u8>,
+    cell: GILOnceCell<PyObject>,
```

Comment:
We now use `GILOnceCell ` instead of `OnceLock` for lazy initialization of the Python UDF object. In this way, we can use `GILOnceCell::get_or_try_init` instead of `OnceLock::get_or_init` for better error propagation. `OnceLock::get_or_try_init` is not available on stable Rust yet.